src/Compilers/Test/Utilities/VisualBasic/TestOptions.vb

```diff
@@ -32,17 +28,26 @@ End Class
 
 Friend Module TestOptionExtensions
     <Extension()>
-    Public Function WithFeature(options As VisualBasicParseOptions, feature As String, value As String) As VisualBasicParseOptions
-        Return options.WithFeatures(options.Features.Concat({New KeyValuePair(Of String, String)(feature, value)}))
-    End Function
-
-    <Extension()>
     Public Function WithStrictFeature(options As VisualBasicParseOptions) As VisualBasicParseOptions
-        Return options.WithFeature("Strict", "true")
+        Return options.WithFeatures(options.Features.Concat(New KeyValuePair(Of String, String)() {New KeyValuePair(Of String, String)("Strict", "true")}))
     End Function
 
     <Extension()>
-    Public Function WithDeterministicFeature(options As VisualBasicParseOptions) As VisualBasicParseOptions
-        Return options.WithFeature("Deterministic", "true")
+    Friend Function WithExperimental(options As VisualBasicParseOptions, ParamArray features As Feature()) As VisualBasicParseOptions
+        If features.Length = 0 Then
+            Throw New InvalidOperationException("Need at least one feature to enable")
+        End If
+
+        Dim list As New List(Of KeyValuePair(Of String, String))
+        For Each feature In features
+            Dim flag = feature.GetFeatureFlag()
+            If flag Is Nothing Then
```

Comment:
Or is this a feature id? I guess the name is confusing somewhat.

---
In reply to: [67381960](https://github.com/dotnet/roslyn/pull/12039#discussion_r67381960) [](ancestors = 67381960)